pkg/resourceprovider/provider.go

```diff
@@ -159,7 +160,14 @@ func (p *resourceProvider) GetPodMetrics(pods ...*metav1.PartialObjectMetadata)
 	resultsByNs := make(map[string][]nsQueryResults, len(podsByNsBatched))
 	for result := range resChan {
 		if result.err != nil {
-			klog.Errorf("unable to fetch metrics for pods in namespace %q, skipping: %v", result.namespace, result.err)
+			if promErr, ok := result.err.(*client.Error); ok {
+				klog.Errorf("prom-adapter unable to fetch metrics for pods in namespace %q, error: %s", result.namespace, promErr.Error())
+				mprom.PodQueryFailureCounter.WithLabelValues(result.namespace, fmt.Sprintf("%d", promErr.StatusCode)).Inc()
+			} else {
+				// Generic error handling for other types of errors
+				klog.Errorf("prom-adapter unable to fetch metrics for pods in namespace %q, error: %s", result.namespace, result.err.Error())
```

Comment:
ditto: nit: remove `prom-adapter` from these comments because they'll be in prom-adapter service logs anyway